src/java/org/apache/cassandra/db/partitions/PartitionUpdate.java

```diff
@@ -624,6 +624,13 @@ public void validateIndexedColumns()
 
     public static class PartitionUpdateSerializer
     {
+        private final SchemaManager schemaManager;
```

Comment:
Let's make one step further - don't keep this big, clumsy `SchemaManager` object.
Let's keep `SchemaProvider` or even a simpler `Function<TableId, TableMetadata>`